examples/pi.c

```diff
@@ -0,0 +1,61 @@
+#include <shmem.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+#include <math.h>
+
+#define NUM_POINTS 10000
+
+long long inside = 0, total = 0;
+
+int
+main(int argc, char* argv[], char *envp[])
+{
+    int me, myshmem_n_pes;
+    /*
+    ** Starts/Initializes SHMEM/OpenSHMEM
+    */
+    shmem_init();
+    /*
+    ** Fetch the number or processes
+    ** Some implementations use num_pes();
+    */
+    myshmem_n_pes = shmem_n_pes();
+    /*
+    ** Assign my process ID to me
+    */
+    me = shmem_my_pe();
+
+    srand(1+me);
+
+    for(total = 0; total < NUM_POINTS; ++total) {
+        double x,y;
+        x = rand()/(double)RAND_MAX;
+        y = rand()/(double)RAND_MAX;
+
+        if(x*x + y*y < 1) {
+            ++inside;
+        }
+    }
+
+    shmem_barrier_all();
+
+    if(me == 0) {
+        for(int i = 1; i < myshmem_n_pes; ++i) {
```

Comment:
Minor comment: Since the user might run this program with older gcc, it might be better to declare `i` at the top before.